src/Agent.Listener/SelfUpdater.cs

```diff
@@ -258,7 +315,12 @@ private async Task DownloadLatestAgent(CancellationToken token)
 
                 if (!downloadSucceeded)
                 {
-                    throw new TaskCanceledException($"Agent package '{archiveFile}' failed after {Constants.AgentDownloadRetryMaxAttempts} download attempts");
+                    string exceptionMessage = $"Agent package '{archiveFile}' failed after {Constants.AgentDownloadRetryMaxAttempts} download attempts.";
+                    if (!_hashValidationDisabled)
```

Comment:
I don't think this is a proper way to display such error. We don't need to display knob advice if there are issues related only to downloading.

Reply:
This means we need to know these two cases separately:
1) if download succeeded
2) if hash validation succeeded